assets/ui/components/ArticleBodyHtml.tsx

```diff
@@ -22,21 +40,51 @@ class ArticleBodyHtml extends React.PureComponent<any, any> {
         // use memoize so this function is only called when `body_html` changes
         this.getBodyHTML = memoize(this._getBodyHTML.bind(this));
 
-        this.bodyRef = React.createRef();
+        this.bodyRef = React.createRef<HTMLDivElement>();
     }
 
+
     componentDidMount() {
-        this.loadIframely();
+        const item = this.props.item;
+        const html = this.getBodyHTML(
+            (item.es_highlight?.body_html ?? '').length > 0 ?
+                item.es_highlight?.body_html[0] :
+                item.body_html
+        );
+
+        if (!html) {
+            return;
+        }
+
+        const prepareWirePreview = extensions.prepareWirePreview ?? ((element) => element);
+        const previewElement = prepareWirePreview(
```

Comment:
We're already processing the HTML in the `_updateImageEmbedSources` function. This function manipulates the HTML to modify the `src` attribute of `img, video, audio` tags. The extension point we need provides similar functionality, to inspect and possibly modify the Article's HTML.

So that we can keep these functionalities grouped together, it might be good to add this extension point to `_getBodyHTML` instead, and use the [HTMLContent](https://github.com/superdesk/newsroom-core/blob/develop/assets/components/HTMLContent.jsx) element in the render function so we are not manipulating the DOM directly but instead letting React do that for us (and so `dompurify` is used). Considering `_getBodyHTML` is memoized, it will only get called **if** the Article's HTML is changed.

Reply:
we can't use react for rendering due to embeds scripts, we can do some workarounds for certain embeds like we do for iframely but there is no generic fix afaik

Reply:
Understood